File: EIDSSAVR/vb/EIDSS/EIDSS.Tests/Database/DummyLookupCacheHelper.cs
using bv.common.db.Core;
using bv.common.Objects;

namespace EIDSS.Tests.Database
{
    public enum DummyLookupTables
    {
        Country,
        Rayon,
        Region,
        Settlement,
        Organization,
        Department,
        Person,
        AvianDiagnoses,
        HumanDiagnoses
    }

    public class DummyLookupCacheHelper
    {

        public static void Init()
        {
            LookupCache.Clear();
            LookupCache.LookupTables.Add(DummyLookupTables.Country.ToString(), new LookupTableInfo(DummyLookupTables.Country, true, null, null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Organization.ToString(), new LookupTableInfo(DummyLookupTables.Organization, true, "tlbOffice", null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Department.ToString(), new LookupTableInfo(DummyLookupTables.Department, true, "tlbOffice", null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Person.ToString(), new LookupTableInfo(DummyLookupTables.Person, true, null, null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Rayon.ToString(), new LookupTableInfo(DummyLookupTables.Rayon, true, null, null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Region.ToString(), new LookupTableInfo(DummyLookupTables.Region, true, null, null, null, null));
            LookupCache.LookupTables.Add(DummyLookupTables.Settlement.ToString(), new LookupTableInfo(DummyLookupTables.Settlement, true, null, null, null, null));
            LookupTableInfo table = new LookupTableInfo(DummyLookupTables.AvianDiagnoses, true, "trtDiagnosis", "spDiagnosis_SelectLookup", null, null);
            table.Parameters.Add("@HACode", 1 | 32 | 64);
            LookupCache.LookupTables.Add(DummyLookupTables.AvianDiagnoses.ToString(), table);
            table = new LookupTableInfo(DummyLookupTables.HumanDiagnoses, true, "trtDiagnosis", "spDiagnosis_SelectLookup", null, null);
            table.Parameters.Add("@HACode", 2);
            LookupCache.LookupTables.Add(DummyLookupTables.HumanDiagnoses.ToString(), table);
            LookupCache.Init(true,true);

        }
    }
}

File: EIDSSAVR/vb/EIDSS/EIDSS_NOTIFICATION_SVC/Tests/DummyLookupCacheHelper.vb
Imports bv.common.db.Core
Imports bv.common.Objects
Namespace Tests
    Public Enum LookupTables
        Country
        Rayon
        Region
        Settlement
        Organization
        Department
        Person
        AvianDiagnoses
        HumanDiagnoses
    End Enum
    Public Class DummyLookupCacheHelper
        Public Shared Sub Init()
            Dim table As LookupTableInfo
            LookupCache.Clear()
            LookupCache.LookupTables.Add(LookupTables.Country.ToString, New LookupTableInfo(LookupTables.Country, True))
            LookupCache.LookupTables.Add(LookupTables.Organization.ToString, New LookupTableInfo(LookupTables.Organization, True, "Office"))
            LookupCache.LookupTables.Add(LookupTables.Department.ToString, New LookupTableInfo(LookupTables.Department, True, "Office"))
            LookupCache.LookupTables.Add(LookupTables.Person.ToString, New LookupTableInfo(LookupTables.Person, True))
            LookupCache.LookupTables.Add(LookupTables.Rayon.ToString, New LookupTableInfo(LookupTables.Rayon, True))
            LookupCache.LookupTables.Add(LookupTables.Region.ToString, New LookupTableInfo(LookupTables.Region, True))
            LookupCache.LookupTables.Add(LookupTables.Settlement.ToString, New LookupTableInfo(LookupTables.Settlement, True))
            table = New LookupTableInfo(LookupTables.AvianDiagnoses, True, "Diagnosis_List", "spDiagnosis_List_HACode_SelectLookup")
            table.Parameters.Add("@HA_Code", 1 Or 32 Or 64)
            LookupCache.LookupTables.Add(LookupTables.AvianDiagnoses.ToString, table)
            table = New LookupTableInfo(LookupTables.HumanDiagnoses, True, "Diagnosis_List", "spDiagnosis_List_HACode_SelectLookup")
            table.Parameters.Add("@HA_Code", 2)
            LookupCache.LookupTables.Add(LookupTables.HumanDiagnoses.ToString, table)
            LookupCache.Init()

        End Sub
    End Class

End Namespace

